src/torchrunx/launcher.py

```diff
@@ -114,6 +104,35 @@ def run(
         :return: A dictionary mapping worker ranks to their output
         :rtype: dict[int, Any]
         """
+
+        logger = logging.getLogger("torchrunx")
+        logger.setLevel(logging.DEBUG)
+        logger.propagate = False
+
+        log_dir = Path(self.log_dir)
+        log_dir.mkdir(parents=True, exist_ok=True)
+        # timestamp = datetime.datetime.now().isoformat(timespec="seconds")
+
+        if self.log_spec is None:
+            # TODO: this assumes the type of workers_per_host is simply int. We should consider
+            # again whether it's worth supporting inhomogeneous allocations (list[int])
+            self.log_spec = default_logging(
+                hostnames=self.hostnames,
+                num_workers=self.workers_per_host,  # type: ignore
+                log_dir=os.fspath(log_dir),
+            )
+
+        log_formatter = logging.Formatter("%(asctime)s:%(levelname)s:%(name)s:%(message)s")
+
+        for lname, handlers in self.log_spec.items():  # type: ignore
+            _logger = logging.getLogger(f"torchrunx.{lname}")
+            for handler in handlers:
+                handler.setFormatter(log_formatter)
+                _logger.addHandler(handler)
+
+        log_process = Process(target=monitor_log, args=(), daemon=True)
```

Comment:
just a nitpick: can we do

`target=LogRecordSocketReceiver(host=socket.getfqdn()).serve_until_stopped()`